src/Listener/EventContext.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+
+namespace SilverStripe\Snapshots\Listener;
+
+
+abstract class EventContext
+{
+    abstract public function getAction(): string;
```

Comment:
```suggestion
    abstract public function getAction(): ?string;
```
This needs to return `?string` as `null` is a valid value representing the case where action could not be determined. All instances where this function is called need to handle `null` value by returning early.